examples/federated-connections/langchain/src/agents/agent.ts

```diff
@@ -0,0 +1,67 @@
+import { AIMessage } from "@langchain/core/messages";
+import { RunnableLike } from "@langchain/core/runnables";
+import {
+  Annotation,
+  END,
+  InMemoryStore,
+  MemorySaver,
+  MessagesAnnotation,
+  START,
+  StateGraph,
+} from "@langchain/langgraph";
+import { ToolNode } from "@langchain/langgraph/prebuilt";
+import { ChatOpenAI } from "@langchain/openai";
+
+import { withCalendarFreeBusyAccess } from "../lib/auth0-ai";
+import { checkCountryHoliday, checkUserCalendar } from "./tools";
+
+const model = new ChatOpenAI({
+  model: "gpt-4o",
+}).bindTools([checkUserCalendar, checkCountryHoliday]);
+
+const callLLM = async (state: typeof MessagesAnnotation.State) => {
+  const response = await model.invoke(state.messages);
+  return { messages: [response] };
+};
+
+const routeAfterLLM: RunnableLike = function (state) {
+  const lastMessage = state.messages[state.messages.length - 1] as AIMessage;
+  if (!lastMessage.tool_calls?.length) {
+    return END;
+  }
+  return "tools";
+};
+
+const StateAnnotation = Annotation.Root({
+  ...MessagesAnnotation.spec,
+});
+
+const stateGraph = new StateGraph(StateAnnotation)
```

Comment:
This example feels very complicated and if I'm a dev new to Auth0 looking at this, it might be a turn off IMO. Can we add a simpler example that shows just using FCAT without requiring so much setup and a langgraph server. This feels like more advanced example for step up auth with resume and such.

Reply:
Agree, it feels like an advance example. We can think about a simpler/basic example on a separate PR once we merge this one.